libbeat/publisher/pipeline/retry.go

```diff
@@ -126,10 +132,12 @@ func (r *retryer) loop() {
 
 		log = r.logger
 	)
+	r.doneWaiter.Add(1)
 
 	for {
 		select {
 		case <-r.done:
+			r.doneWaiter.Done()
```

Comment:
Although it's the only location we return from, better move it to the beginning of the method and use `defer r.doneWaiter.Done()`.